tests/containers/buildah_image.pm

```diff
@@ -22,30 +22,46 @@ use containers::common;
 use containers::container_images;
 use containers::urls 'get_suse_container_urls';
 use version_utils qw(get_os_release check_os_release);
+use version_utils 'is_sle';
 
 sub run {
     my ($image_names, $stable_names) = get_suse_container_urls();
     my ($running_version, $sp, $host_distri) = get_os_release;
-
     install_buildah_when_needed($host_distri);
     install_podman_when_needed($host_distri);
+    install_docker_when_needed($host_distri);
+    allow_selected_insecure_registries(runtime => 'docker');
     allow_selected_insecure_registries(runtime => 'podman');
     scc_apply_docker_image_credentials() if (get_var('SCC_DOCKER_IMAGE'));
 
     for my $iname (@{$image_names}) {
         test_container_image(image => $iname, runtime => 'buildah');
         if (check_os_release('suse', 'PRETTY_NAME')) {
             # sle15-working-container is the default name given to a container. it is created in test_container_image
-            test_opensuse_based_image(image => 'sle15-working-container', runtime => 'buildah');
+            my $prefix_img_name = +(split /\//, +(split /:/, $iname)[0])[-1];
+            test_opensuse_based_image(image => "${prefix_img_name}-working-container", runtime => 'buildah');
             # Due to the steps from the test_opensuse_based_image previously,
             # the image has been committed as refreshed
-            test_containered_app(runtime => 'buildah',
+            test_containered_app(runtime => 'podman',
+                buildah    => 1,
+                dockerfile => 'Dockerfile.suse',
+                base       => 'refreshed');
+            assert_script_run "podman rmi -f myapp";
+            test_containered_app(runtime => 'docker',
+                buildah    => 1,
                 dockerfile => 'Dockerfile.suse',
                 base       => 'refreshed');
+            if (is_sle) {
+                my $container_id = script_output "docker ps -aqf 'ancestor=myapp'";
+                assert_script_run "docker cp /etc/SUSEConnect $container_id:/etc/SUSEConnect";
+                assert_script_run "docker cp /etc/zypp/credentials.d/SCCcredentials $container_id:/etc/zypp/credentials.d/SCCcredentials";
+                assert_script_run "docker exec -it $container_id zypper lr";
```

Comment:
The `/etc/SUSEConnect ` should not be in the container and the `/etc/zypp/credentials.d/SCCcredentials` is automatically linked from the container host. This logic is also already done in `test_opensuse_based_image()` subroutine.

Reply:
i tested it and it failed
i had done my research a few week ago and i quite sure for the approach
i just cant remember where i found out or if i came up with it trying out things
but for sure the image that it was built from the dockerfile fails to run zypper if you do not cp the files
i dont know and i dont remember if that was something that it was required some extra step in the dockerfile to avoid it

Reply:
I'll summarise our today's finding in the main thread of this pull request.